File: languages/js/src/entity/authenticity/signature_header_jws.ts
import * as proto from "../../bridge/proto/identity_entities";

/**
 * Represents the header of a JSON Web Signature (JWS). [RFC 7515](https://datatracker.ietf.org/doc/html/rfc7515).
 */
export class SignatureHeaderJws {
  alg: string;
  kid: string;

  /**
   * Constructs a SignatureHeaderJws object with the specified algorithm and key identifier.
   * @param alg 
   * @param kid 
   */
  constructor(alg: string, kid: string) {
    this.alg = alg;
    this.kid = kid;
  }

  public static fromProto(s: proto.SignatureHeaderJWS): SignatureHeaderJws {
    return new SignatureHeaderJws(s.alg, s.kid);
  }

  toProto(): proto.SignatureHeaderJWS {
    return proto.SignatureHeaderJWS.fromPartial({
      alg: this.alg,
      kid: this.kid
    });
  }
}

File: languages/go/entity/authenticity/signature_header_jws.go
package authenticity

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// SignatureHeaderJws represents the header of a JSON Web Signature (JWS). [RFC 7515].
//
// [RFC 7515]: https://datatracker.ietf.org/doc/html/rfc7515
type SignatureHeaderJws struct {
	Alg string
	Kid string
}

func NewSignatureHeaderJwsFromProto(s *proto.SignatureHeaderJWS) SignatureHeaderJws {
	if s == nil {
		return SignatureHeaderJws{}
	}
	return SignatureHeaderJws{
		Alg: s.Alg,
		Kid: s.Kid,
	}
}

func (s SignatureHeaderJws) ToProto() *proto.SignatureHeaderJWS {
	return &proto.SignatureHeaderJWS{
		Alg: s.Alg,
		Kid: s.Kid,
	}
}
